src/services/cache/ByURLAsyncCache.ts

```diff
@@ -13,6 +14,9 @@ export class ByURLAsyncCache<TValue>
     this._promiseMap = new URLMap<Promise<TValue>>();
   }
 
+  private readonly _onDidInvalidate = new vscode.EventEmitter<URL>();
+  readonly onDidInvalidate = this._onDidInvalidate.event;
```

Comment:
I had added this and then ended up swapping the instance using it for a `SerializedKeyMap` where I added the `onDidChange`. I just left it since it seemed something less to implement later if needed. I'm fine to remove it if you'd prefer though.